extensions/replication/queueProcessor/task.js

```diff
@@ -239,8 +239,28 @@ function initAndStart(zkClient) {
             }
         });
 
+        /**
+         * Get probe config will pull the configuration for the probe server based on
+         * the provided site key, and if the probe server config is not an array, it will
+         * return the global probe server config.
+         *
+         *  Get the probe server config for the first site in the site names list,
+         *  as if the probeConfig is an array there is only one element in site names
+         *
+         * @param {Object} queueProcessorConfig - Configuration of the queue processor that
+         *      holds the probe server configs for all sites
+         * @param {Array<String>} siteNames - List of site names
+         * @returns {ProbeServerConfig} Config for site or global config
+         */
+        function getProbeConfig(queueProcessorConfig, siteNames) {
```

Comment:
this function should not be in this "bootstrap" script : there is already too much here, let's not add to the mess.
and this will allow testing this function (even we do not test -yet- the startup script itself)